feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
+
+        if self.missing_values == "raise":
+            # check if dataset contains na
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        # Split the dataframe into a basis and a measurement dataframe.
+        basis_df, measurement_df = self._split_dataframe(X)
+
+        # Switch base and measurement dataframe if required.
+        if self.switch_basis:
+            measurement_df, basis_df = basis_df, measurement_df
+
+        # Compute the PSI
+        self.psi = self._compute_PSI(basis_df, measurement_df, self.bucketer)
+
+        # Select features below the threshold
+        self.features_to_drop_ = self.psi[
+            self.psi.value >= self.threshold
+        ].index.to_list()
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def _compute_PSI(self, df_basis, df_meas, bucketer):
+        """
+        Compute the PSI by first bucketing (=binning) and then call the psi calculator
+        for each feature.
+
+        Parameters
+        ----------
+        df_basis : pandas dataframe of shape = [m_rows, n_features]
+            Serve as reference for the features distribution.
+
+        df_meas : pandas dataframe of shape = [p_rows, n_features]
+            Matrix for which the feature distributions will be compared to the basis.
+
+        bucketer : EqualFrequencyDiscretiser or EqualWidthDiscretiser.
+                Default = EqualFrequencyDiscretiser
+            Class used to bucket (bin) the features in order to approximate
+            the distribution.
+
+        Returns
+        -------
+        results_df: pandas dataframe.
+            Dataframe containing the PSI for each feature.
+        """
+        # Initialize a container for the results.
+        results = {}
+
+        # Compute the PSI for each feature excluding the column used for split.
+        for feature in filter(lambda x: x != self.split_col, self.variables_):
+            results[feature] = [
+                self._compute_feature_psi(df_basis[[feature]], df_meas[[feature]])
+            ]
+
+        # Transform the result container in a user friendly format.
+        results_df = pd.DataFrame.from_dict(results).T
+        results_df.columns = ["value"]
+
+        return results_df
+
+    def _compute_feature_psi(self, series_basis, series_meas):
+        """
+        Call the PSI calculator for two distributions
+
+        Parameters
+        ----------
+        series_basis : pandas series
+            Proxy for the reference distribution.
+
+        series_meas : pandas series
+            Proxy for the measurement distribution.
+
+        Returns
+        -------
+        psi_value: float.
+            PSI value.
+        """
+        # Perform the binning for all features.
+        basis_binned = (
+            self.bucketer.fit_transform(series_basis.dropna()).value_counts().fillna(0)
```

Comment:
series_basis.dropna() is dropping the na from the original data. But we are giving the user options to raise include or ignore, should not capture that functionality here?

Reply:
Good point. In the current case, include does not make much sense. I will remove the 'include' option for missing.